### PR TITLE
[CLOSING > See: #1081] feat(rule): css-orientation-lock new wcag21 rule

### PR DESCRIPTION
**Pull request by jkodu**
**To keep commit history clean, opened a new PR https://github.com/dequelabs/axe-core/pull/1081 for the same. Multiple attempts were made to fix CI build**

New rule: 
- WCAG 2.1 rule
- Rule Id: `css-orientation-lock`
- Description: Ensures that the content is not locked to any specific display orientation, and functionality of the content is operable in all display orientations (portrait/ landscape).

Closes Issue:
- https://github.com/dequelabs/axe-core/issues/851

## Reviewer checks

**Required fields, to be filled out by PR reviewer(s)**
- [ ] Follows the commit message policy, appropriate for next version
- [ ] Has documentation updated, a DU ticket, or requires no documentation change
- [ ] Includes new tests, or was unnecessary
- [ ] Code is reviewed for security by: << Name here >>

### CODE REVIEW

**Comment by jkodu** [comment]
Closing this for now, as other CSSOM work(s) needs to be merged and changes reflected into this branch.

**Comment by jkodu** [comment]
Tidying up. Closing PR, will re-open when work on this rule is started again.

**Comment by jkodu** [comment]
Re-opening this to begin preliminary work.

There is a dependency on PR - https://github.com/dequelabs/axe-core/pull/1000
Once that is merged, this rule can be authored.

**Comment by jkodu** [comment]
Closing this PR - until https://github.com/dequelabs/axe-core/pull/1000 is merged, as there is a very close dependency & cannot progress.

File: lib/rules/css-orientation-lock.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+	"id": "css-orientation-lock",
+	"selector": "html",
+	"tags": [
+		"cat.structure",
+		"wcag262",
+		"wcag21aa"
+	],
+	"metadata": {
+		"description": "Ensures content is not locked to any specific display orientation, and the content is operable in all display orientations",
+		"help": "Content is operable in all display orientations"
```
Comment (WilcoFiers) [inline lib/rules/css-orientation-lock.json]: I think this needs to be more generic. Something along the lines of "CSS Media queries are not used to lock display orientation"?

File: test/checks/mobile/css-orientation-lock.js [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+						sheet: getSheet(
+							SHEET_DATA.MEDIA_STYLE_ORIENTATION_WITH_TRANSFORM_ROTATE_90
+						)
+					}
+				]
+			}
+		);
+		assert.isFalse(actual);
+	});
+
```
Comment (WilcoFiers) [inline test/checks/mobile/css-orientation-lock.js]: Should test the related nodes.

Comment (jkodu) [reply test/checks/mobile/css-orientation-lock.js]: Done in integration tests below.

Comment (WilcoFiers) [reply test/checks/mobile/css-orientation-lock.js]: Unit tests are intended to test the logic of your code. Use integration tests to make sure all of the pieces work together. Tests are better isolated that way.

Comment (jkodu) [reply test/checks/mobile/css-orientation-lock.js]: The reason related nodes are tested in integration as against unit, is I could not  find an example of how to get relatedNodes from check results with in unit tests. Any pointers on that will be appreciated.

**Comment by jkodu** [comment]
@WilcoFiers 
Review again.

The `selenium-webdriver` integration tests were failing for any page that is dependent on external resources, and analysis led to the following issues:

1) CORS not enabled, when requesting `css` from `cdn`. This was later ruled out, as `js` files from `cdn` was fetched normally. I ensured that by inspecting via fiddler the network data when running `circleci` locally & the responses are 200, but the resources were not 100% loaded at all times (see point 3 below).
2) Ordering of styles and scripts in respective html pages, which led to blocking scenario for page load.
3) The webdriver setup does not have mechanism to handle wait for actual page to finish loading. (see - https://www.seleniumhq.org/docs/03_webdriver.jsp#fetching-a-page). 

The fixes put in place involves catering to the above edge cases (not a 100%).
Currently looking at enhancing webdriver to be smart enough around external resources.

**Comment by jkodu** [comment]
@WilcoFiers - So CI is behaving (https://circleci.com/gh/dequelabs/axe-core/tree/new-rule-css-orientation-lock)

**Comment by jkodu** [comment]
**To keep commit history clean, opened a new PR https://github.com/dequelabs/axe-core/pull/1081 for the same. Multiple attempts were made to fix CI build**

**Pull request closed without merging**